packages/dds/tree/src/test/rebaserAxiomaticTests.ts

```diff
@@ -380,6 +654,196 @@ function verifyComposeAssociativity<TChangeset>(
+	}
+}
+
+function verifyRebaseLeftDistributivity<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	namedEditsToRebaseOver: NamedChangeset<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(namedEditsToRebaseOver.length === 2, "expected at least 2 edits");
+	const editsToRebaseOver = namedEditsToRebaseOver.map(({ changeset }) => changeset);
+	const revInfo = defaultRevInfosFromChanges(editsToRebaseOver);
+
+	const editB = editsToRebaseOver[0];
+	const editC = editsToRebaseOver[1];
+	const rebaseMetaData = rebaseRevisionMetadataFromInfo(revInfo, [
+		editB.revision,
+		editC.revision,
+	]);
+	const actualChange = makeAnonChange(
+		fieldRebaser.rebaseComposed(rebaseMetaData, edit.change, editB, editC),
+	);
+
+	const expectedChange = fieldRebaser.rebase(fieldRebaser.rebase(edit.change, editB), editC);
+
+	assertDeepEqual(actualChange, makeAnonChange(expectedChange));
+}
+
+function verifyRebaseRightDistributivity<TChangeset>(
+	edits: TaggedChange<TChangeset>[],
+	namedEditToRebaseOver: NamedChangeset<TChangeset>,
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(edits.length === 2, "expected 2 edits");
+	const editToRebaseOver = namedEditToRebaseOver.changeset;
+	const metadata = defaultRevisionMetadataFromChanges([editToRebaseOver]);
+	const editA = edits[0];
+	const editB = edits[1];
+	const editC = {
+		change: editToRebaseOver.change,
+		revision: namedEditToRebaseOver.changeset.revision,
+		metadata,
+	};
+
+	// (A ○ B) ↷ C
+	const actualChange = rebaseTagged(
+		fieldRebaser.rebase,
+		makeAnonChange(fieldRebaser.compose(editA, editB)),
+		[editC],
+	);
+
+	const editRebasedOverC = rebaseTagged(fieldRebaser.rebase, editA, [editC]);
+	const invertedEdit = tagRollbackInverse(fieldRebaser.invert(editA), undefined, editA.revision);
+	// (A ↷ C) ○ (B ↷ (A⁻¹ ○ C ○ (A ↷ C)))
+	const expectedChange = makeAnonChange(
+		fieldRebaser.compose(
+			editRebasedOverC,
+			rebaseTagged(fieldRebaser.rebase, editB, [
+				makeAnonChange(
+					fieldRebaser.compose(
+						makeAnonChange(fieldRebaser.compose(invertedEdit, editC)),
+						editRebasedOverC,
+					),
+				),
+			]),
+		),
+	);
+
+	assertDeepEqual(actualChange, expectedChange);
+}
+
+function verifyRebaseOverUndoRedoPair<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	namedEditsToRebaseOver: NamedChangeset<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(namedEditsToRebaseOver.length === 1, "expected just 1 edit");
+	const editsToRebaseOver = namedEditsToRebaseOver.map(({ changeset }) => changeset);
+	const editB = { change: editsToRebaseOver[0].change, revision: editsToRebaseOver[0].revision };
+
+	const inverseEditB = fieldRebaser.invert(editB);
+
+	// ((A ↷ B) ↷ B⁻¹) ↷ B
+	const actualChange = makeAnonChange(
```

Comment:
It makes more sense to tag this change with `edit.revision`.

Reply:
Tagged change with `edit.revision`